image.go

```diff
@@ -156,8 +157,17 @@ func (i *image) Labels() map[string]string {
 }
 
 func (i *image) RootFS(ctx context.Context) ([]digest.Digest, error) {
+	if i.diffIDs != nil {
+		return i.diffIDs, nil
+	}
+
 	provider := i.client.ContentStore()
-	return i.i.RootFS(ctx, provider, i.platform)
+	diffIDs, err := i.i.RootFS(ctx, provider, i.platform)
+	if err != nil {
+		return nil, err
+	}
+	i.diffIDs = diffIDs
+	return diffIDs, nil
```

Comment:
This is not thread-safe.  `i.diffIDs` needs to be protected by a mutex.

Reply:
ok, do it right away